### DISCUSSION TITLE
Cannot edit repository description when no code has been provided

### DISCUSSION THREAD

**Discussion by marckruzik** [community/community#43623] [Repositories]
As far as I know, the only way to edit a repository description on GitHub website is to use the "about" button next to the repository description.

Here is how the "about" button normally appears:
![IMG_20230108_141121.jpg](https://user-images.githubusercontent.com/51711029/211197802-28272da2-b7fd-432f-80b7-f4a5e801a928.jpg)

But when no code has been provided, the repository main page does not display the about section:
![IMG_20230108_135222.jpg](https://user-images.githubusercontent.com/51711029/211197868-8960e70c-b4ac-4729-b562-ca21f7fb2f8d.jpg)

Therefore, because of this special "no code has been provided" page, we cannot access the about section, and so we have no way to edit the repository description.

I think that in this case, the about section should be displayed, and we should be able to edit the repository description.

How to reproduce:
Create an empty repository using GitHub website.

**Comment by ghostinhershell** [comment]
Hey there!  👋

Thanks for posting in the GitHub Community, @marckruzik! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

I've gone ahead and moved this to the correct category for you so that your feedback can get visibility. Good luck!

**Reply by marckruzik** [reply]
Thanks ghostinhershell!

**Comment by marckruzik** [comment]
Hello, this bug is still here, and it's still confusing people.
I agree this is not a big problem, and there are workarounds (such as adding any file in the repo, by example by using the link "creating a new file", so the repo is no longer empty). But this bug occurs in **the first minute** following the creation of an empty repo, which gives a bad impression.

Today, when promoting GitHub (and Aspire) and showing a new user how to use GitHub, we created an empty repo, then tried to add a description. Which does not work, due to the bug discussed here.
As I remember creating an issue one year and half ago, I'm coming again here, hoping to get some attention.

**Comment by mtsknn** [comment]
One workaround, which I just tested and confirmed to work, is to use [GitHub CLI](https://github.com/cli/cli):

```sh
gh repo edit --description="Bacon ipsum"
```

(Found this via [Stack Overflow](https://stackoverflow.com/a/70455907/1079869).)

**Reply by MrSuddenJoy** [reply]
@mtsknn many thanks. Had no idea `GitHub` has `cli` tool.......

**Comment by aflacake** [comment]
Thanks to open issue on discussion. New users need the best UX

**Reply by MrSuddenJoy** [reply]
@aflacake Agreed. But question still stands: what this has to do with problem at hand?

**Reply by aflacake** [reply]
Visit issue: https://github.com/orgs/community/discussions/54372#discussioncomment-11359353

**Comment by 19barsav** [comment]
This is really annoying when creating several template repositories for a classroom. I do not have the assignments or code ready to upload, but I do need to link them to my course management software, and the links are different for the ones with/without code because I cannot add a description. I cannot add a topic to organize my course from another's in our organization, either. It's been over two years and there still hasn't been a fix. 

My only 'fix' is to add a 'readme' file every time and leave it blank, but this would be really inconvenient if I wasn't uploading the files directly.

**Reply by MrSuddenJoy** [reply]
> this would be really inconvenient if I wasn't uploading the files directly.

@19barsav why this approach would be inconvenient?
